libraries/AP_Scripting/examples/smbus-check-cycles.lua

```diff
@@ -0,0 +1,21 @@
+-- This script checks SMBus battery cycle count
+
+local warning_cycles = 100
+local battery_instance = 0
+
+function update()
+  if not arming:is_armed() then -- only run check when disarmed
+    local cycle_count = battery:get_cycle_count(battery_instance)
+    if cycle_count then
+      if cycle_count >= warning_cycles then
+        gcs:send_text(0, string.format("Battery needs replacing (%d cycles)", cycle_count))
```

Comment:
Something I realized I did people a disservice with on some examples, you could also make the string here by concatanation:

```lua
gcs:send_text(0, "Battery needs replacing (" .. cycle_count .. " cycles)")
```

It's really your choice which you use, I just wanted to note that some people find concatenation easier if they aren't doing any control of the number printing. (On the other hand sometimes its easier to read a format string to know what the output looks like)

Nothing needs to be changed here, I just wanted to note it.

Reply:
ok, thanks for the advice, I'll leave it as it is I think.